domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -2224,51 +2221,206 @@ public void getDistinctOn(TestContext context) throws IOException {
       });
     async.awaitSuccess();
 
-    String whereClause =  "WHERE jsonb->>'order_format' = 'Other'";
-    Async async2 = context.async();
+    String whereClause = "WHERE jsonb->>'order_format' = 'Other'";
     postgresClient.get(MOCK_POLINES_TABLE, Object.class, "*", whereClause, false, false,
       false, null, distinctOn, handler -> {
         ResultInfo resultInfo = handler.result().getResultInfo();
         context.assertEquals(1, resultInfo.getTotalRecords());
-        async2.complete();
+        async.complete();
```

Comment:
You cannot reuse an Async. The second awaitSuccess() will not wait until the second assertEquals has been executed.